[Table] Colored definition table shows overlapping border in top left cell

lubber-de opened the pull request:
##  Description
A colored `definition table` was still showing the table color border on the top left cell (which is not used on definition tables by design)
I first thought it was enough to change the background, but IE/Edge still showed the border on zooming, thats why i still had to use and adjust the box-shadow

## Testcase
https://jsfiddle.net/gc15zfku/

## Screenshots
### Before
![883_evil](https://user-images.githubusercontent.com/18379884/61477775-4fc1da80-a990-11e9-9a62-b90f37810b2e.gif)

### Using only background (FYI)
![883_bad](https://user-images.githubusercontent.com/18379884/61477795-610ae700-a990-11e9-9bc6-549bcad0b57c.gif)

### Final Fix
![883_good](https://user-images.githubusercontent.com/18379884/61477811-6700c800-a990-11e9-83e6-06ea91a2fcea.gif)

## Closes
#883 
https://github.com/Semantic-Org/Semantic-UI-React/issues/3710

exoego commented:
If some issues are found only in Firefox, it might be a bug in Firefox...
I suggest to report an issue in
https://bugzilla.mozilla.org/home
You can now login tonthr issue tracker with GitHub account.

lubber-de commented:
So glad, i now found a fix for firefox without the need for a `legacy` workaround, so i removed that one again by 0e6de97a5c8c03ca714f9bf3c33d76612f390e68 😃 
Adjusted jsfiddle https://jsfiddle.net/gc15zfku/
